### ISSUE TITLE
[Offline support] Load project list from local db when offline

### ISSUE THREAD

**Issue by gino-m** [google/ground-android#306]
Currently, if the user user clicks "Join project" when offline, the project list is empty.

**Comment by gino-m** [comment]
Hi @shobhitagarwal1612! This is an extension of the offline project sync that you implemented - care take a look at this one? An alternative would be to show a message saying "can't change projects when offline", but it seems easy enough to fall back to local db for the project list. Wdyt?

**Comment by shobhitagarwal1612** [comment]
This looks like a regression. I tested this feature while working on the offline sync

**Comment by shobhitagarwal1612** [comment]
The problem is that the `FirebaseFirestoreException` is not getting thrown on making remote calls

**Comment by gino-m** [comment]
I see. Can we fall back to the local db on any exception for now?

**Comment by shobhitagarwal1612** [comment]
The problem is that no exception is being thrown at all. So, `doOnError` and `onErrorResumeNext` don't get called

**Comment by gino-m** [comment]
Perhaps it's getting thrown, but getting eaten somewhere upstream? Can you try logging `doOnError` as for upstream as possible? (e.g., in our `GndFirestore`class?)

**Comment by shobhitagarwal1612** [comment]
Digging deeper into the source from where the issue gets generated. Adding `doOnError` in `GndFirestore` class also didn't help.

**Comment by gino-m** [comment]
You're absolutely right. The stack trace above wasn't from the actual loading of the feature, not the project list.

It appears that Firestore SDK's `Query#get()`, which is used by `RxFirestore#getCollection()`, returns an empty `QuerySnapshot` instead of an error. 

I tried to force an error by invoking `ref.getFirestore().enableNetwork()`, but that doesn't fail immediately; sometimes it keeps trying until the 10 s timeout is reached, and sometimes it just returns success even though the network isn't available. 😱

For good measure I verified that `FirestoreSettings` were still set to `persistenceEnabled=false` at the call site, so either this is a bug or we're misusing the API. We should create a min repro to post to https://github.com/firebase/firebase-android-sdk/issues.

In the meantime, I'm preparing a workaround that forces an error in this case if network is not available.
